tests/components/intellifire/conftest.py

```diff
@@ -17,35 +41,241 @@ def mock_setup_entry() -> Generator[AsyncMock]:
+        new_callable=AsyncMock,
+    ) as mock_connectivity:
+        mock_connectivity.return_value = (False, False)
+        yield mock_connectivity
+
+
+@pytest.fixture
+def mock_connectivity_test_fail_fail_then_pass_pass() -> (
+    Generator[AsyncMock, None, None]
+):
+    """Mock both Cloud and Local Connectivity."""
+    return_values = [(False, False), (True, True), (True, True), (True, True)]
+
+    with patch(
+        "homeassistant.components.intellifire.__init__.UnifiedFireplace.async_validate_connectivity",
+        new_callable=AsyncMock,
+    ) as mock_connectivity:
+        mock_connectivity.side_effect = return_values
+        yield mock_connectivity
+
+
+@pytest.fixture
+def mock_poll_local_fireplace_exception() -> Generator[AsyncMock, None, None]:
+    """Mock polling a fireplace that isn't a fireplace."""
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireAPILocal.poll",
+        new_callable=AsyncMock,
+    ) as mock_poll:
+        mock_poll.side_effect = ConnectionError
+        yield mock_poll
+
+
+@pytest.fixture
+def mock_fireplace_finder_none() -> Generator[None, MagicMock, None]:
     """Mock fireplace finder."""
     mock_found_fireplaces = Mock()
     mock_found_fireplaces.ips = []
     with patch(
-        "homeassistant.components.intellifire.config_flow.AsyncUDPFireplaceFinder.search_fireplace"
+        "homeassistant.components.intellifire.config_flow.UDPFireplaceFinder.search_fireplace"
     ):
         yield mock_found_fireplaces
 
 
 @pytest.fixture
-def mock_fireplace_finder_single() -> Generator[MagicMock]:
+def mock_config_entry_current() -> MockConfigEntry:
+    """Return a mock config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        version=1,
+        data={
+            CONF_IP_ADDRESS: "192.168.2.108",
+            CONF_USERNAME: "grumpypanda@china.cn",
+            CONF_PASSWORD: "you-stole-my-pandas",
+            CONF_SERIAL: "3FB284769E4736F30C8973A7ED358123",
+            CONF_WEB_CLIENT_ID: "FA2B1C3045601234D0AE17D72F8E975",
+            CONF_API_KEY: "B5C4DA27AAEF31D1FB21AFF9BFA6BCD2",
+            CONF_AUTH_COOKIE: "B984F21A6378560019F8A1CDE41B6782",
+            CONF_USER_ID: "52C3F9E8B9D3AC99F8E4D12345678901FE9A2BC7D85F7654E28BF98BCD123456",
+        },
+        options={CONF_READ_MODE: API_MODE_LOCAL, CONF_CONTROL_MODE: API_MODE_CLOUD},
+        unique_id="serial",
+    )
+
+
+@pytest.fixture
+def mock_config_entry_old() -> MockConfigEntry:
+    """For migration testing."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        version=1,
+        title="Fireplace 3FB284769E4736F30C8973A7ED358123",
+        data={
+            CONF_HOST: "192.168.2.108",
+            CONF_USERNAME: "grumpypanda@china.cn",
+            CONF_PASSWORD: "you-stole-my-pandas",
+            CONF_USER_ID: "52C3F9E8B9D3AC99F8E4D12345678901FE9A2BC7D85F7654E28BF98BCD123456",
+        },
+    )
```

Comment:
I'd recommend only creating this in the tests you need this, because the naming scheme with `_old` `super_old` is going to be annoying real quick

Reply:
super old is no more... we don't need it ... so should I leave it or move it to the test now?